docs/docs/creating-a-source-plugin.md

```diff
@@ -106,15 +90,19 @@ required fields. This includes automatically generating fields like node IDs
 and the `contentDigest` MD5 hash, keeping your code focused on data gathering,
 not boilerplate.
 
+## Gotcha: don't forget to return!
+
+After your plugin is finished sourcing nodes, it should either return a promise or use the callback (3rd parameter) to report back to Gatsby when `sourceNodes` is fully executed. If a promise or callback isn't returned, Gatsby will continue on in the build process, before nodes are finished being created. Your nodes might not end up in the generated schema at compilation, or the process will hang while waiting for an indication that it's finished.
```

Comment:
maybe some examples here? or links to code snippets to plugins that do "to return a promise or callback"

Reply:
A future person can add these if desired -- I'm not the expert on this part & just moved that text around.